src/ssh-util/src/tunnel.rs

```diff
@@ -64,8 +64,9 @@ impl Default for SshTimeoutConfig {
 /// Specifies an SSH tunnel.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct SshTunnelConfig {
-    /// The hostname of the SSH bastion server.
-    pub host: String,
+    /// The hostname/IP of the SSH bastion server.
+    /// If multiple hosts are specified, they are tried in order.
+    pub host: Vec<String>,
```

Comment:
We use this as a key when sharing ssh tunnels through the same bastion. To ensure we don't over-connect, this should be a `BTreeSet` so its sorted. Actually, we should sort in the pg and mysql connection code too.

Note that there are weird cases now where:
- a host resolves to a different number of ip's on different workers
- different workers connect to different upstream ips for the pg/mysql/etc server, if they appear connectable at different times

Both cases result in additional ssh connections, which is not something we can work around, but might be worth a comment

Reply:
good call out! Switching to a `BtreeSet` across all connection types and will note those edge cases in a comment